src/couch_db_updater.erl

```diff
@@ -1454,3 +1468,44 @@ default_security_object(_DbName) ->
         "everyone" ->
             []
     end.
+
+% These functions rely on using the process dictionary. This is usually frowned
+% upon howver in this case it is done to avoid changing to a different server state
+% record. Once PSE (Pluggable Storage Engine) code lands this should be moved to the
+% #db{} record.
+
+schedule_timeout() ->
+    case get(idle_timer_ref) of
+        TimerRef when is_reference(TimerRef) ->
+            erlang:cancel_timer(TimerRef);
+        undefined  ->
+            ok
+    end,
+    case idle_limit() of
+        infinity ->
+            ok;
+        Timeout when is_integer(Timeout) ->
+            TRef = erlang:send_after(Timeout, self(), timeout),
```

Comment:
The timeout is due to the hibernate since hibernation is only ended by receiving a message. Ideally we'd have a {hibernate, timeout()} reply option but that doesn't exist.